src/main/java/jp/co/soramitsu/sora/crypto/json/flattener/FlattenedKeyParser.java

```diff
@@ -0,0 +1,113 @@
+package jp.co.soramitsu.sora.crypto.json.flattener;
+
+import static java.lang.Integer.parseUnsignedInt;
+import static java.lang.String.format;
+import static java.util.regex.Pattern.compile;
+import static jp.co.soramitsu.sora.crypto.json.flattener.Flattener.ARRAY_DELIMITER;
+import static jp.co.soramitsu.sora.crypto.json.flattener.Flattener.DICT_DELIMITER;
+import static jp.co.soramitsu.sora.crypto.json.flattener.TokenType.COLON;
+import static jp.co.soramitsu.sora.crypto.json.flattener.TokenType.NUMBER;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class FlattenedKeyParser {
+
+  @NonNull
+  private String stream;
+
+  private String consumeString(int length) throws ParsingException {
+    return consume(compile(format(".{%d}", length)));
+  }
+
+  private Integer consumeInt() throws ParsingException {
+    try {
+      Pattern pattern = NUMBER.getPattern();
+      String consumed = consume(pattern);
+      return parseUnsignedInt(consumed);
+
+    } catch (NumberFormatException e) {
+      throw new ParsingException(e);
+    }
+  }
+
+  private String consumeToken(TokenType token) throws ParsingException {
+    return consume(token.getPattern());
+  }
+
+  private String parseDictKey() throws ParsingException {
+    consumeToken(TokenType.DICT_DELIMITER);
```

Comment:
static import

Reply:
Not possible, because this file contains other DICT_DELIMITER